sdk/security_keyvault/examples/delete_secret.rs

```diff
@@ -1,24 +1,19 @@
-use azure_identity::{ClientSecretCredential, TokenCredentialOptions};
+use azure_identity::DefaultAzureCredentialBuilder;
 use azure_security_keyvault::SecretClient;
-use std::env;
-use std::sync::Arc;
+use std::{env, sync::Arc};
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let client_id = env::var("CLIENT_ID").expect("Missing CLIENT_ID environment variable.");
-    let client_secret =
-        env::var("CLIENT_SECRET").expect("Missing CLIENT_SECRET environment variable.");
-    let tenant_id = env::var("TENANT_ID").expect("Missing TENANT_ID environment variable.");
     let keyvault_url =
         env::var("KEYVAULT_URL").expect("Missing KEYVAULT_URL environment variable.");
     let secret_name = env::var("SECRET_NAME").expect("Missing SECRET_NAME environment variable.");
 
-    let creds = Arc::new(ClientSecretCredential::new(
-        tenant_id,
-        client_id,
-        client_secret,
-        TokenCredentialOptions::default(),
-    ));
+    let creds = Arc::new(
+        DefaultAzureCredentialBuilder::new()
+            .exclude_managed_identity_credential()
```

Comment:
If this is always excluded in our examples, should we perhaps think about inverting this and requiring users who want it, to opt into it?

Reply:
see #1016